scripts/system/create/entitySelectionTool/entitySelectionTool.js

```diff
@@ -31,6 +33,55 @@ function deepCopy(v) {
     return JSON.parse(JSON.stringify(v));
 }
 
+function getDuplicateAppendedName(originalName, forcedIncrement) {
+    var duplicateName, existingNames;
+    var delayCounter = forcedIncrement;
+    var duplicateNumber = 1;
+    var rippedOriginalName = removeNameAppendice(originalName);
+
+    do {
+        duplicateNumber++;
+                if (rippedOriginalName === "") {
+                    duplicateName = "(" + duplicateNumber + ")";
+                } else {
+                    duplicateName = rippedOriginalName + " (" + duplicateNumber + ")";
+                }
+                existingNames = Entities.findEntitiesByName(duplicateName, Vec3.ZERO, ENTIRE_DOMAIN_SCAN_RADIUS, false);
```

Comment:
I think this is at the core of why I don’t love this solution.  1. it’s not a good idea to call this method in a loop (it can be expensive, it’s blocking, and this loop could be huge), 2. what if you’ve got Box, Box (2), and Box (3), and you delete Box (2), and then copy Box again?

Reply:
What other solution do we have?
a sequence per name kept for the session?

Or simply not doing this functionality.
For sure, this can rapidly become an annoying feature when used for repetitive cloning.

Reply:
keeping track of it per name won’t work because the script could be reloaded or stopped

like I said in the issue, I would vote to have Box always become Box (2), even if you clone it multiple times